src/aliceVision/numeric/numeric.hpp

```diff
@@ -570,6 +570,13 @@ void SplitRange(const T range_start, const T range_end, const int nb_split,
   }
 }
 
+template<class T>
+T divideRoundUp(T x, T y)
+{
+    static_assert(std::is_integral<T>::value, "divideRoundUp only works with integer arguments");
+    return (x + y - 1) / y;
```

Comment:
this does not work for negative values though:
```c++
int main()
{
    std::cout << std::fixed
              << "ceil(+2.4) = " << std::ceil(+2.4) << " " << divideRoundUp(24, 10) <<'\n'
              << "ceil(-2.4) = " << std::ceil(-2.4) << " " << divideRoundUp(-24, 10) << '\n'
              << "ceil(-2.4) = " << std::ceil(-2.4) << " " << divideRoundUp(24, -10) << '\n';
}
```
you get
```
ceil(+2.4) = 3.000000    3
ceil(-2.4) = -2.000000    -1
ceil(-2.4) = -2.000000    -1
```
I know I'm a lazy guy ;-), but wouldn't be easier to just make a wrapper around `std::ceil()` casting to float the arguments for the division and returning a cast to `T`m like
```c++
return static_cast<T>(std::ceil((float)x, (float)y));
```
I know, a lot of waste but... :-)

Reply:
Indeed, negative numbers explicitly wasn't what I wanted to support. I should have added a comment.

However, doing something like `static_cast<T>(std::ceil((float)x / (float)y))` suffers from precision problems, because float is only precise up to 2^24, which is something like 16 million.

Reply:
I've fixed the negative number problem now by using slightly different way to compute the division result.